proj-android/PowerAuthLibrary/src/main/java/io/getlime/security/powerauth/sdk/impl/CancelableAsynchronousOperation.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2018 Wultra s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getlime.security.powerauth.sdk.impl;
+
+import android.support.annotation.NonNull;
+
+import java.util.concurrent.Semaphore;
+
+import io.getlime.security.powerauth.networking.interfaces.IAsyncOperation;
+import io.getlime.security.powerauth.networking.interfaces.ICancelable;
+
+/**
+ * The {@code CancelableAsynchronousOperation} class allows execution of {@link IAsyncOperation}
+ * on the background thread. The operation itself also implements {@link ICancelable} interface and
+ * therefore the same instance is provided as parameter to the {@link IAsyncOperation#onExecution(ICancelable)}.
+ *
+ * @see ICancelable
+ * @see IAsyncOperation
+ */
+public class CancelableAsynchronousOperation implements Runnable, ICancelable {
```

Comment:
I guess we shouldn't handle this situation.
If anyone wants to make something async within a runnable, make it their responsibility.